src/librustc/ty/layout.rs

```diff
@@ -541,6 +545,9 @@ pub struct Struct {
     pub memory_index: Vec<u32>,
 
     pub min_size: Size,
+
+    /// Size of padding needed by the struct due to requested repr alignment
+    pub padding: Vec<Size>,
```

Comment:
Can't this be computed on-demand based on the sizes of the fields and the alignments we can get out of LLVM types? Keep in mind constant enum variant values *already* have to deal with recomputing padding based on the values at hand, I'd rather generalize that algorithm than have this vector.